### PR TITLE
ARROW-9159: [Python] Implement Array.isnull/isvalid methods

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Instead of having the NotImplementedError

### CODE REVIEW

File: python/pyarrow/tests/test_compute.py [hunk 1]
```diff
@@ -313,3 +313,16 @@ def test_compare_chunked_array_mixed():
         arr_chunked == arr_chunked2,
     ]:
         assert result.equals(expected)
+
+
+def test_isnull():
+
```
Comment (pitrou) [inline python/pyarrow/tests/test_compute.py:319]: Any reason for the empty line here?

**Review by pitrou — APPROVED** [review]
LGTM

File: python/pyarrow/tests/test_compute.py (hunk 1 above)
Comment (jorisvandenbossche) [reply python/pyarrow/tests/test_compute.py]: No specific reason, no, removed

**Comment by wesm** [comment]
Can you add isvalid while you are at it?

**Comment by jorisvandenbossche** [comment]
Added `isvalid` as well. Two more questions:

- Do we rather prefer the underscore version (`is_valid`, `is_null`)? Since the existing `isnull` wasn't working yet, I don't think we necessarily need to keep the name for the reason that it already existed. 
  Personally, I generally prefer the `_` versions (and which makes it also consistent with the name in the function registry), although for `isnull` the resemblance with pandas is nice as well.
- Do I also add them as functions in `pyarrow.compute`?

**Comment by wesm** [comment]
Let's do the underscores version. Maybe people will curse us but "a foolish consistency is the hobgoblin of little minds"

**Comment by wesm** [comment]
I'm making the change quickly

**Review by wesm — APPROVED** [review]
+1, merge on green

**Comment by jorisvandenbossche** [comment]
Thanks for finishing this!

**Pull request closed without merging**